packages/cursorless-engine/src/processTargets/modifiers/getContainingScopeTarget.ts

```diff
@@ -36,8 +36,8 @@ export function getContainingScopeTarget(
       scope = expandFromPosition(
         scopeHandler,
         editor,
-        scope.domain.end,
-        "forward",
+        scope.domain.start,
```

Comment:
This change was necessary for python where the parent statement does not exist after each child due to the lack of a closing delimiter. See the python test for example that did not work without this change.

Reply:
That fixes this specific case, but still leaves us with the same bug if we ever have scopes that start at the same position. I think you could iterate forward from the end using `allowAdjacentScopes: true` and `containment: "required"`, and then just count scopes that contain the current scope until you reach `ancestorIndex`. Does that make sense?

Reply:
Made a slight update now. Not sure if this is correct, but passes all the test.

Reply:
I believe this approach will break down if cursor is between adjacent scopes. Let's say scopes look like the following (document text is `0123`):

```
   0123
a:   - 
b:    - 
c:  -- 
d:  --- 
```

And cursor is between `2` and `3` and scope preference breaks left, so containing scope is `a`. If you say `"grand grand scope"`, you will get `b`, when you should get `d`. I think you need to keep a `currentScope`, initialize it to the original containing scope, filter out scopes that are not a parent of the current scope, and keep updating it and increment ancestor count each time you find a matching scope. Not sure if that makes sense?

Reply:
Do you actually have a a code example of this I can try?

Reply:
No this is the kinda thing you'd want to use a mock scope for, like in that base target test I sent you

Reply:
I can do that if you want to. At the same time if we can't actually come up with a real world code example that doesn't work isn't it good enough...at least for now?

Reply:
https://github.com/cursorless-dev/cursorless/issues/2138